### PR TITLE
nuget: 6.3.1.1 -> 6.6.1.2, move out of dotnetPackages

### PR DESCRIPTION
**Pull request by mdarocha**
## Description of changes

Helps with #234775

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@NixOS/dotnet

### CODE REVIEW

**Comment by mdarocha** [comment]
Result of `nixpkgs-review pr 255386` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>msbuild</li>
    <li>networkminer</li>
    <li>nuget</li>
  </ul>
</details>

**Comment by raphaelr** [comment]
This PR has a merge conflict with the master branch at the moment.

**Comment by jiridanek** [comment]
Looks legit to me, however

I guess I should probably get myself removed as a maintainer, since I no longer use NixOS on my machines and over time I've forgotten the little of nixpkgs language I used to know.

**Comment by mdarocha** [comment]
The conflict should be resolved now

File: pkgs/by-name/nu/nuget/package.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    rev = "upstream/${attrs.version}.bin";
+    sha256 = "sha256-9/dSeVshHbpYIgGE/8OzrB4towrWVB3UxDi8Esmbu7Y=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  installPhase = ''
+    mkdir -p $out/lib/${attrs.pname}
```
Comment (raphaelr) [inline pkgs/by-name/nu/nuget/package.nix:19]: Missing a `runHook preInstall` at the beginning, and a `runHook postInstall` at the end

**Review by raphaelr — COMMENTED** [review]
Result of `nixpkgs-review pr 255386` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>nuget (dotnetPackages.Nuget)</li>
    <li>msbuild</li>
    <li>networkminer</li>
  </ul>
</details>

---

LGTM apart from the installPhase nit

File: pkgs/by-name/nu/nuget/package.nix (hunk 1 above)
Comment (mdarocha) [reply pkgs/by-name/nu/nuget/package.nix:21]: done

**Review by raphaelr — APPROVED** [review]

**Comment by dev-null-undefined** [comment]
Doesn't this PR break `aarch64-linux` compatibility?
If so should it?

**Comment by dev-null-undefined** [comment]
@mdarocha any updates?

**Comment by mdarocha** [comment]
> @mdarocha any updates?

https://github.com/NixOS/nixpkgs/pull/264619

**Pull request merged**